model2vec/distill/tokenizer.py

```diff
@@ -109,7 +119,17 @@ def add_tokens(tokenizer: Tokenizer, tokens_to_add: list[str]) -> Tokenizer:
                 wordpiece_vocab[token] = len(wordpiece_vocab)
 
     elif model == "Unigram":
-        raise ValueError("Adding tokens to a unigram tokenizer is not supported.")
+        pre_tokenizer = data["pre_tokenizer"]
+        if pre_tokenizer['type'] != 'Metaspace':
```

Comment:
This if does not have an accompanying else with a useful error message. So if the pre-tokenizer _is_ `Metaspace`, we just continue as if everything worked, but it didn't. I also don't immediately get why check if itsn't Metaspace, this should be an `==`, right? 

Also, as a general point, consider using guards instead of nested ifs, e.g.:

```python
if not x:
    raise ValueError("")
if not y:
   raise ValueError("")
```

This makes the code easier to parse.